feat: add cli support to clear default cache

deependujha opened the pull request:
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lit-data/blob/main/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes #626 

<img width="1440" alt="Screenshot 2025-06-28 at 5 17 26 PM" src="https://github.com/user-attachments/assets/bd8aa452-9b73-4b12-b9ac-6fef2e1d8660" />


## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

deependujha commented:
@robmarkcole can you try this pr, and let me know if it helps in clearing cache. Next I can focus on profiling.

One thing to note is: it can only list/delete default cache and not your specified cache directory for SD.

Borda reviewed (changes requested):
if we do CLI let's have consistent experience also with litServe: https://github.com/Lightning-AI/LitServe/blob/main/src/litserve/cli.py

bhimrazy commented on requirements.txt:
```diff
@@ -1,5 +1,6 @@
 torch
 torchvision
+typer
```
Let’s avoid including it in the main requirements. If needed, we can add it under extras.

deependujha replied:
removed typer to be consistent with `litserve`

bhimrazy reviewed (commented):
IMO, a CLI might not be necessary here just to clear the cache—unless there are additional plans for it that I'm missing.

deependujha commented on src/litdata/cli/commands.py:
```diff
@@ -0,0 +1,70 @@
+        required=True,
+        help="Path to the dataset to optimize.",
+    )
+    optimize_parser.set_defaults(func=lambda args: print(f"Optimizing dataset at {args.dataset}..."))
+
+
```
It's just a placeholder for now, we may include more subcommands and arguments in subsequent prs.

deependujha commented:
@Borda I've updated the CLI and removed the `typer` dependency. The structure now follows the same pattern as the `litserve` CLI.

> One difference I noticed: **LitServe** makes use of `lightning_sdk` helpers to run certain commands. As far as I know, there aren't similar utility functions in `lightning_sdk` for `litdata` yet, but please correct me if I'm wrong.

Other than that, the CLI structure and style are now consistent across both projects.

Borda commented:
lets psl update the PR description with examples :)
